Frameworks/Accelerate/vImage.cpp

```diff
@@ -951,4 +935,876 @@ vImage_Error vImageMatrixMultiply_ARGB8888(const vImage_Buffer* src,
+    const uint32_t minByteAlignedBytesPerRow = ((minBitsPerRow + 7) & ~(7)) >> 3;
+    const uint32_t alignedPitch = _vImageAlignUInt(minByteAlignedBytesPerRow, c_vImageBlockSizeBytes);
+    vImage_Error returnCode = kvImageNoError;
+
+    if ((flags & ~(kvImageNoFlags | kvImageNoAllocate | kvImagePrintDiagnosticsToConsole | kvImageDoNotTile)) != 0) {
+        TraceWarning(TAG, L"Unsupported flag(s)");
+        return kvImageUnknownFlagsBit;
+    }
+
+    FAIL_FAST_IF_FALSE(height > 0 && width > 0 && bitsPerFragment > 0);
+
+    buffer->height = height;
+    buffer->width = width;
+
+    if ((flags & kvImageNoAllocate) == kvImageNoAllocate) {
+        // Align rowBytes
+        buffer->rowBytes = alignedPitch;
+        buffer->data = nullptr;
+        returnCode = c_vImageBlockSizeBytes;
+    } else if (c_vImagePadAllocs == true) {
+        // For 4bytesPerFragment pixels, SSE2 instructions operate on 4 pixels at a time
+        const uint32_t minCompilerAlignmentBytes = 4;
+        const uint32_t additionalRowPaddingBytes = (c_vImageBlockSizeBytes - minCompilerAlignmentBytes);
+        const uint32_t maxPitch = alignedPitch + additionalRowPaddingBytes;
+        const uint32_t allocSize = maxPitch * height;
+
+        // Note: We can't use alignedMalloc here since the client is responsible for freeing the buffer and can use free or aligned_free
+        buffer->data = malloc(allocSize);
+
+        if (buffer->data != nullptr) {
+            // SIMD load/store instructions operate best on 16byte aligned memory and in blocks of 16bytes
+            // Padding to ensure that for unaligned start addresses, at the very least:
+            // 1. A single 16pixel block can be operated on per row
+            // 2. Every second row will be 16byte aligned
+            const uint32_t bytePaddingForIdealAlignment =
+                (c_vImageBlockSizeBytes - ((uint32_t)(buffer->data) & (c_vImageBlockSizeBytes - 1))) % c_vImageBlockSizeBytes;
+            buffer->rowBytes = alignedPitch + bytePaddingForIdealAlignment;
+        } else {
+            // Try again without additional padding
+            buffer->rowBytes = alignedPitch;
+            buffer->data = malloc(buffer->rowBytes * height);
+
+            if (buffer->data == nullptr) {
+                returnCode = kvImageMemoryAllocationError;
+            }
+        }
+    } else {
+        buffer->rowBytes = alignedPitch;
+        buffer->data = malloc(buffer->rowBytes * height);
+
+        if (buffer->data == nullptr) {
+            returnCode = kvImageMemoryAllocationError;
+        }
+    }
+
+    return returnCode;
+}
+
+// Alias vImageConvert_ARGB8888toPlanar8 with vImageConvert_BGRA8888toPlanar8 and vImageConvert_RGBA8888toPlanar8
+_vImageFunctionInterfaceCvt8888ToPlanar8 vImageConvert_BGRA8888toPlanar8 = vImageConvert_ARGB8888toPlanar8;
+_vImageFunctionInterfaceCvt8888ToPlanar8 vImageConvert_RGBA8888toPlanar8 = vImageConvert_ARGB8888toPlanar8;
```

Comment:
It won't because the inputs will also be swapped (ie: the interface to the function is vImageConvert_RGBA8888toPlanar8(redPlaneBuffer, greenPlaneBuffer, bluePlaneBuffer, alphaPlaneBuffer) )

The function vImageConvert_ARGB8888toPlanar8 interleaves the planes in the same order as the parameters are passed in so if the inputs order is changed, the component order in the resulting pixel buffer will change with it.